### PR TITLE
fix:[PLG-287]Single Shipper configuration changes pom.xml

### PR DESCRIPTION
**Pull request by AnandPaladin**
# Description

This change is to make all shipper jobs pointing to same jar file(data-shipper.jar) though job Name will be different for each cloud soruce
Testing details:
https://paladincloud.atlassian.net/browse/PLG-287?focusedCommentId=15869
https://docs.google.com/document/d/1DZi-2nw4pl75Mcb79CH7VEi8DYddD1uY2i9oFkEOgRw/edit

Fixes # (issue)

Changing Shipper pom.xml to generate data-shipper.jar in CE repo

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also
list any relevant details for your test configuration

- [x] Test A
Detailed steps are documented in Jira Ticket. Built the data-shipper.jar from local and deployed manually at SAASDEV and ran the individual shipper jobs by editing custom event bridge pattern.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] My commit message/PR follows the contribution guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# **Other information**:

List any documentation updates that are needed for the Wiki

### CODE REVIEW

**Review by rnithinpaladin — APPROVED** [review]

**Comment by AnandPaladin** [comment]
> As for the test result - I was expecting a bit different results and approach.
> 
> * You showing GCP even rule, but then run redhat and AWS. Not consistent.
> * When you test if RedHat will work with new approach - have you renamed/removed `data-shipper-redhat.jar` file or it was there all the time?
> * As for job params - I expect them to exactly opposite from what you've shared.
>   This is what you have shared.
> 
> ```
> ["~/fetch_and_run.sh","data-shipper-redhat.jar","{\"jobName\": \"data-shipper\", \"jobUuid\": \"data-shipper\", \"jobType\": \"jar\", \"jobDesc\": \"Ship redhat Data from S3 to Paladin ES\", \"params\": [{\"encrypt\": false, \"key\": \"package_hint\", \"value\": \"com.tmobile.cso.pacman\"}, {\"encrypt\": false, \"key\": \"config_creds\", \"value\": \"dXNlcjpwYWNtYW4=\"}, {\"encrypt\": false, \"key\": \"datasource\", \"value\": \"redhat\"}, {\"encrypt\": false, \"key\": \"s3.data\", \"value\": \"mapper-results/saasdev/redhat/20231010-125030\"}]}","-Xms1024m -Xmx4g","policy-engine.jar","com.tmobile.pacman.executor.JobExecutor"]
> ```
> 
> And this is what I would expect:
> 
> ```
> ["~/fetch_and_run.sh","data-shipper.jar","{\"jobName\": \"data-shipper-redhat\", \"jobUuid\": \"data-shipper-redhat\", \"jobType\": \"jar\", \"jobDesc\": \"Ship redhat Data from S3 to Paladin ES\", \"params\": [{\"encrypt\": false, \"key\": \"package_hint\", \"value\": \"com.tmobile.cso.pacman\"}, {\"encrypt\": false, \"key\": \"config_creds\", \"value\": \"dXNlcjpwYWNtYW4=\"}, {\"encrypt\": false, \"key\": \"datasource\", \"value\": \"redhat\"}, {\"encrypt\": false, \"key\": \"s3.data\", \"value\": \"mapper-results/saasdev/redhat/20231010-125030\"}]}","-Xms1024m -Xmx4g","policy-engine.jar","com.tmobile.pacman.executor.JobExecutor"]
> ```
> 
> * How that first parameter of the job is used? I'm talking about `data-shipper-redhat.jar`. Have you checked the logs?
>   What I see there is that `data-shipper-redhat.jar` file was used. It means that you never removed/renamed it, and that `data-shipper.jar` file you created never been used.
>   You can see it by yourself, [this is the log](https://us-east-1.console.aws.amazon.com/cloudwatch/home?region=us-east-1#logsV2:log-groups/log-group/saasdev%2Fjobs/log-events/saasdev%2Fdefault%2Fe858a2a6579443fbb8383bfddb23d800) from the job you started.
> 
> This is example of proper [job](https://us-east-1.console.aws.amazon.com/batch/home?region=us-east-1#jobs/detail/77611eca-4773-4e9d-bf92-5d1d95745ed3) because it uses Please redo the tests and update results. and before the test remove/rename all other shipper jar files so you will be 100% sure that if it runs ok then it uses only single data shipper file

@mkinchin-paladin Please ignore previous test results there steps were very cluttered. Please see the attached doc for detailed steps and let me know if you have any queries after following steps mentioned in this doc.
https://docs.google.com/document/d/1DZi-2nw4pl75Mcb79CH7VEi8DYddD1uY2i9oFkEOgRw/edit
[All.Shipper.Jobs.Testing-UUID.data-shipper.jar.pdf](https://github.com/PaladinCloud/CE/files/12882524/All.Shipper.Jobs.Testing-UUID.data-shipper.jar.pdf)

**Review by plyubich — DISMISSED** [review]

File: jobs/pacman-data-shipper/pom.xml [hunk 1]
```diff
@@ -221,9 +221,8 @@
 						<configuration>
 							<target>
 								<copy file="target/data-shipper-jar-with-dependencies.jar" tofile="../../dist/jobs/data-shipper-jar-with-dependencies.jar"/>
-								<copy file="target/data-shipper-jar-with-dependencies.jar" tofile="../../dist/jobs/data-shipper-azure.jar"/>
-								<copy file="target/data-shipper-jar-with-dependencies.jar" tofile="../../dist/jobs/data-shipper-gcp.jar"/>
-								<copy file="target/data-shipper-jar-with-dependencies.jar" tofile="../../dist/jobs/data-shipper-redhat.jar"/>
+								<copy file="target/data-shipper-jar-with-dependencies.jar" tofile="../../dist/jobs/data-shipper.jar"/>
+
```
Comment (mkinchin-paladin) [inline jobs/pacman-data-shipper/pom.xml:225]: Redundant empty line

**Review by mkinchin-paladin — CHANGES_REQUESTED** [review]
Tests look good, but I don't see that you applied those scheduler event changed anywhere.
It must be changed as well so this PR will not break a thing.

File: jobs/pacman-data-shipper/pom.xml (hunk 1 above)
Comment (AnandPaladin) [reply jobs/pacman-data-shipper/pom.xml]: Removed these lines in latest push
@santhosh-challa as per your review comments removed unused jar from pom.xml and formatted the code and removed empty line

**Review by santhosh-challa — DISMISSED** [review]
LGTM

**Comment by santhosh-challa** [comment]
@AnandPaladin Please do other installation changes needed for OSS. For EE these should be enough.

**Review by plyubich — DISMISSED** [review]
Where are the changes to the events that pass the filename?

Since now we have a single shipper, the lambda that creates the shipper batch should just simply start the generic shipper since the shipper filename is now a single constant. Now the events don't need to contain any shipper related metadata.

**Comment by AnandPaladin** [comment]
> Where are the changes to the events that pass the filename?
> 
> Since now we have a single shipper, the lambda that creates the shipper batch should just simply start the generic shipper since the shipper filename is now a single constant. Now the events don't need to contain any shipper related metadata.

@plyubich  There is going to three shipper jobs only thing is they all can make use of the same jar(data-shipper.jar) to execute their shipper job. So job name will be distinct for all of them only job UUID will remain the same. keeping same job name will cause the problem of identifying which shipper job has failed for which data source which we have identified during testing. Hence decided to keep job name distinct keeping job UUID the same.
cc @mkinchin-paladin @santhosh-challa

File: jobs/pacman-data-shipper/pom.xml [hunk 2]
```diff
@@ -232,7 +230,7 @@
 					</execution>
 				</executions>
 			</plugin>
-        </plugins>
-   </build>
+  </plugins>
+  </build>
```
Comment (mkinchin-paladin) [inline jobs/pacman-data-shipper/pom.xml:234]: Revert

**Review by mkinchin-paladin — CHANGES_REQUESTED** [review]
JobUUID_Change-Azure_Event_Pattern
@AnandPaladin please test changes in this PR once again. We need all these changes in CE, as well as in EE repo (right after this PR is merged).
When you test -  you need to modify events correspondingly and remove/rename/move other shipper files (like data-shipper-redhat.jar) from saasdev folder so they will not intervene you test results.

@mkinchin-paladin Tested them by renaming the existing shipper jar files and tested by sending event bridge event by changing job UUID to **data-shipper** . Please find the test screen shots for the same.
Renaming_All_Shipper_Jar

**Review by mkinchin-paladin — APPROVED** [review]
LGTM

**Pull request merged**